pkg/knowledge_base/edge_kb.go

```diff
@@ -404,6 +407,8 @@ func (kb EdgeKB) findPathsInGraph(source, dest core.Resource, stack []graph.Edge
 	return result
 }
 
+// GetTrueUpstream takes in a resource and returns all upstream resources which exist in the dag, if their edge does not specify the reverse direction flag.
```

Comment:
```suggestion
// GetTrueUpstream takes in a resource and returns all upstream resources which exist in the dag, if their edge does not specify the reverse direction flag in the knowledge base.
```
nit: I know this makes sense in context of being a method on EdgeKB, but this helps make it clear why this function is here as opposed to on the ResourceGraph.